mitodl/mit-open
# add to list dialog updates

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5275

### Description (What does it do?)
This PR alters the Add to List dialog used to add Learning Resources to both Learning Paths and User Lists. Instead of adding the resource to lists immediately upon checking the box, the user's changes are saved upon clicking the new "Save" button. New API endpoints were added to accomplish this, sending all the lists that the resource should be a part of as one request. The styling of the modal was also updated to reflect the current designs in Figma.

### Screenshots (if appropriate):
![image](https://github.com/user-attachments/assets/8eae98c5-25a3-4499-9c4a-e4b5284f57b3)
![image](https://github.com/user-attachments/assets/3ca778aa-5e19-4c52-a3fe-12a5a2b0d34a)

### How can this be tested?
 - Spin up `mit-learn` on this branch
 - Make sure you are logged in as a user with `is_staff` or `is_superuser` set to true
 - Make sure you have backpopulated at least some data into your local database and generated a search index that will give you some search results
 - Go to the search page at http://localhost:8063/search
 - On any search result, click the bookmark button to display the Add to User List modal
 - Use the Create New List button to create a few lists
 - Check some boxes and click the x in the upper right of the modal
 - Verify that your changes were not saved
 - Open the modal again and this time select a few lists and hit the save button
 - Verify that the bookmark icon on the resource you selected is now highlighted
 - When you click the bookmark icon again, your selections should come back up
 - Repeat this process with the list button next to the bookmark button, which should open the Add to Learning Path dialog
